kafka-client/src/test/java/dev/responsive/kafka/integration/RowLevelTtlIntegrationTest.java

```diff
@@ -0,0 +1,218 @@
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class RowLevelTtlIntegrationTest {
+
+  @RegisterExtension
+  static ResponsiveExtension EXTENSION = new ResponsiveExtension(StorageBackend.CASSANDRA);
+
+  private static final String INPUT_TOPIC = "input";
+  private static final String OUTPUT_TOPIC = "output";
+  private static final String STORE_NAME = "store";
+
+  private static final Duration DEFAULT_TTL = Duration.ofSeconds(5);
+
+  private final Map<String, Object> responsiveProps = new HashMap<>();
+
+  private String name;
+  private Admin admin;
+
+  @BeforeEach
+  public void before(
+      final TestInfo info,
+      final Admin admin,
+      @ResponsiveConfigParam final Map<String, Object> responsiveProps
+  ) {
+    // add displayName to name to account for parameterized tests
+    name = info.getDisplayName().replace("()", "");
+
+    this.responsiveProps.putAll(responsiveProps);
+
+    this.admin = admin;
+    createTopicsAndWait(admin, Map.of(inputTopic(), 2, outputTopic(), 1));
+  }
+
+  @AfterEach
+  public void after() {
+    admin.deleteTopics(List.of(inputTopic(), outputTopic()));
+  }
+
+  private String inputTopic() {
+    return name + "." + INPUT_TOPIC;
+  }
+
+  private String outputTopic() {
+    return name + "." + OUTPUT_TOPIC;
+  }
+
+  @Test
+  public void test() throws Exception {
+    // Given:
+    final Map<String, Object> properties = getMutableProperties();
+    final KafkaProducer<Long, Long> producer = new KafkaProducer<>(properties);
+    try (final ResponsiveKafkaStreams streams = buildStreams(properties)) {
+      startAppAndAwaitRunning(Duration.ofSeconds(10), streams);
+      for (int i = 0; i < 10000; i++) {
+        pipeInput(inputTopic(), 2, producer, System::currentTimeMillis, 0, 10, 0, 1);
+        Thread.sleep(100);
+      }
+
+      final var kvs = readOutput(outputTopic(), 0, 20, true, properties);
+      assertThat(
+          kvs,
+          hasItems(new KeyValue<>(0L, 10L), new KeyValue<>(1L, 10L)));
+
+    }
+  }
+
+  private ResponsiveKafkaStreams buildStreams(final Map<String, Object> properties) {
+    final StreamsBuilder builder = new StreamsBuilder();
+
+    final KStream<String, String> input = builder.stream(inputTopic());
+    input.process(new TtlProcessorSupplier(), STORE_NAME)
+        .to(outputTopic());
+
+    return new ResponsiveKafkaStreams(builder.build(), properties);
+  }
+
+  private static class TtlProcessorSupplier implements ProcessorSupplier<String, String, String, String> {
+
+    @Override
+    public Processor<String, String, String, String> get() {
+      return new TtlProcessor();
+    }
+
+    @Override
+    public Set<StoreBuilder<?>> stores() {
+      return Collections.singleton(ResponsiveStores.keyValueStoreBuilder(
+          ResponsiveStores.keyValueStore(
+              ResponsiveKeyValueParams.fact(STORE_NAME).withTtlProvider(
+                  TtlProvider.withDefaultTtl(DEFAULT_TTL).withKeyBasedTtl(k -> ttlForKey(k))
```

Comment:
nit: we can probably just have `withDefault` and `withKeyBased` to avoid some verbosity

Reply:
SGTM. Though while we're on the subject, I don't really love the `withKeyBased`/`.withValueBased`/etc method names. Especially if we strip away the "ttl" suffix, it just sounds awkward. Any suggestions?

Reply:
maybe `fromKey`? so `TtlProvider.withDefault(DEFAULT_TTL).fromKey(k -> k.ttl)`